lib/src/code_generator/objc_interface.dart

```diff
@@ -370,16 +364,12 @@ class $name extends ${superType?.name ?? '_ObjCWrapper'} {
   bool _needsConverting(Type type) =>
       type is ObjCInterface ||
       type is ObjCBlock ||
-      _isObject(type) ||
-      _isInstanceType(type);
+      type is ObjCObjectPointer ||
+      type is ObjCInstanceType;
 
   String _getConvertedType(Type type, Writer w, String enclosingClass) {
-    if (type is BooleanType) return 'bool';
-    if (type is ObjCInterface) return type.name;
-    if (type is ObjCBlock) return type.name;
-    if (_isObject(type)) return 'NSObject';
-    if (_isInstanceType(type)) return enclosingClass;
-    return type.getDartType(w);
+    if (type is ObjCInstanceType) return enclosingClass;
```

Comment:
Nice cleanup!

This special case is the `instancetype` right? The one that returns the type of the object itself even in subclasses.